File: lib/features/map/domain/entities/room.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'room.freezed.dart';
part 'room.g.dart';

@freezed
class Room with _$Room {
  const factory Room({
    required String id,
    @Default('') String code, 
    required String fullCode, // C1.1.201
    @Default('') String shortCode, 
    @Default('C1.1') String sector,
    required String title,
    required int floor,
    required String building, // '1', '2'
    required double? lat,
    required double? lng,
    String? teacherId,
    String? description,
    @Default(true) bool isActive,
  }) = _Room;

  factory Room.fromJson(Map<String, dynamic> json) => _$RoomFromJson(json);
}

File: server/models/Room.js
const mongoose = require('mongoose');

const RoomSchema = new mongoose.Schema({
    shortCode: {
        type: String,
        required: [true, 'Короткий код обязателен (например, 201)'],
        trim: true
    },
    sector: {
        type: String,
        required: [true, 'Сектор обязателен (например, C1.1)'],
        trim: true
    },
    fullCode: {
        type: String,
        required: [true, 'Полный код обязателен (например, C1.1.201)'],
        trim: true,
        unique: true
    },
    // Backwards compatibility
    code: {
        type: String,
        trim: true
    },
    building: {
        type: String,
        required: [true, 'Корпус обязателен (например, 1)'],
        trim: true,
        default: 'C1'
    },
    floor: {
        type: Number,
        required: [true, 'Этаж обязателен'],
        min: 1,
        max: 5
    },
    description: {
        type: String,
        trim: true
    },
    // Position as percentage of original map image (0.0 to 1.0)
    positionX: {
        type: Number,
        default: 0.0,
        min: 0.0,
        max: 1.0
    },
    positionY: {
        type: Number,
        default: 0.0,
        min: 0.0,
        max: 1.0
    },
    type: {
        type: String,
        enum: ['LECTURE', 'PRACTICE', 'LAB', 'OFFICE', 'OTHER'],
        default: 'LECTURE'
    },
    isActive: {
        type: Boolean,
        default: true
    }
}, {
    timestamps: true
});

module.exports = mongoose.model('Room', RoomSchema);
